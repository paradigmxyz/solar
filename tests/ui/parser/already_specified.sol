contract C
    is A
    is B //~ ERROR: base contracts already specified
    layout at 0
    layout at 1 //~ ERROR: storage layout already specified
{
    function f()
    public
    private //~ ERROR: visibility already specified

    view
    pure //~ ERROR: state mutability already specified

    virtual
    virtual //~ ERROR: virtual already specified

    override
    override //~ ERROR: override already specified
    {}

    uint
    public
    private //~ ERROR: visibility already specified

    constant
    immutable //~ ERROR: mutability already specified

    virtual //~ ERROR: `virtual` is not allowed here

    override
    override //~ ERROR: override already specified
    x = 0;
}
