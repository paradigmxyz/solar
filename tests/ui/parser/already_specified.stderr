error: base contracts already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     is A
   |        - previous definition
LL |     is B
   |        ^

error: storage layout already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     layout at 0
   |     ----------- previous definition
LL |     layout at 1
   |     ^^^^^^^^^^^

error: visibility already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     public
   |     ------ previous definition
LL |     private
   |     ^^^^^^^

error: state mutability already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     view
   |     ---- previous definition
LL |     pure
   |     ^^^^

error: virtual already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     virtual
   |     ------- previous definition
LL |     virtual
   |     ^^^^^^^

error: override already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     override
   |     -------- previous definition
LL |     override
   |     ^^^^^^^^

error: visibility already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     private
   |     ^^^^^^^

error: mutability already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     immutable
   |     ^^^^^^^^^

error: `virtual` is not allowed here
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     virtual
   |     ^^^^^^^

error: override already specified
  --> ROOT/tests/ui/parser/already_specified.sol:LL:CC
   |
LL |     override
   |     ^^^^^^^^

error: aborting due to 10 previous errors

